### PR TITLE
Backport patches to 8.5.0 and release 33.1

### PR DESCRIPTION
**Pull request by ondrejbudai**
This PR backports the following PRs to the rhel-8.5.0 branch:

- https://github.com/osbuild/osbuild-composer/pull/1702
- https://github.com/osbuild/osbuild-composer/pull/1723
- https://github.com/osbuild/osbuild-composer/pull/1715
- https://github.com/osbuild/osbuild-composer/pull/1762
- https://github.com/osbuild/osbuild-composer/pull/1733
- https://github.com/osbuild/osbuild-composer/pull/1662

There are two commits on top of the backported commits: The first one disables all tests that are not 8.5 nor 9.0. The second one is a release commit for osbuild-composer-33.1

This pull request includes:

- [x] adequate testing for the new functionality or fixed issue
- [x] adequate documentation informing people about the change such as
  - [x] create a file in [news/unreleased](https://github.com/osbuild/osbuild-composer/tree/main/docs/news/unreleased) directory if this change should be mentioned in the release news
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

### CODE REVIEW

**Review by atodorov — APPROVED** [review]

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -38,34 +38,13 @@ RPM:
...
-          - aws/fedora-34-aarch64
-          - aws/rhel-8-x86_64
-          - aws/rhel-8-aarch64
-          - aws/centos-stream-8-x86_64
-          - aws/centos-stream-8-aarch64
       - RUNNER:
           - aws/rhel-8.5-x86_64
           - aws/rhel-8.5-aarch64
           - aws/rhel-9.0-beta-nightly-x86_64
           - aws/rhel-9.0-beta-nightly-aarch64
```
Comment (jrusz) [inline .gitlab-ci.yml:45]: Do we want rhel-9 runners here? I guess just 8.5 are enough, right? Same with base tests section.

**Review by jrusz — COMMENTED** [review]
Just one small comment.

File: osbuild-composer.spec [hunk 2]
```diff
@@ -102,7 +102,7 @@ Obsoletes: osbuild-composer-koji <= 23
 # package which has a different API than the older ones. Let's make the auto-
 # generated code compatible by applying some sed magic.
 #
-# Remove when F33 is EOL
+# Remove when F33.1 is EOL
```
Comment (thozza) [inline osbuild-composer.spec]: I'll bet that you used regex to append `.1` after `33` 😄

File: osbuild-composer.spec [hunk 3]
```diff
@@ -348,7 +378,7 @@ Requires:   python3-lxml
 Requires:   httpd
 Requires:   mod_ssl
 Requires:   openssl
-# see https://bugzilla.redhat.com/show_bug.cgi?id=1986333
+# see https://bugzilla.redhat.com/show_bug.cgi?id=198633.13
```
Comment (thozza) [inline osbuild-composer.spec]: Regexes are hard 😄 . The added `.1` should not be there...

**Review by thozza — CHANGES_REQUESTED** [review]
LGTM, found just cosmetic issue in the SPEC file changes...

**Review by thozza — DISMISSED** [review]
LGTM

File: .gitlab-ci.yml (hunk 1 above)
Comment (thozza) [reply .gitlab-ci.yml:45]: AFAIK at least some changes should go into 9.0 Beta, so this is actually beneficial to test on el9 as well...

**Comment by jrusz** [comment]
One other small thing I remembered, this https://github.com/osbuild/osbuild-composer/commit/b8ad702ee7ea3f79ad64b8d6b8b56983335cc2c9 helps to run the CI faster, but it's not really important so I'm not sure if you want to include it now.

**Comment by ondrejbudai** [comment]
We decided to backport some more test-related PRs:

- https://github.com/osbuild/osbuild-composer/pull/1751
- https://github.com/osbuild/osbuild-composer/pull/1748
- https://github.com/osbuild/osbuild-composer/pull/1745
- https://github.com/osbuild/osbuild-composer/pull/1732
- https://github.com/osbuild/osbuild-composer/pull/1688
- https://github.com/osbuild/osbuild-composer/pull/1675
- https://github.com/osbuild/osbuild-composer/pull/1664
- https://github.com/osbuild/osbuild-composer/pull/1749

**Review by thozza — APPROVED** [review]
Still good

**Comment by ondrejbudai** [comment]
Force-pushed once again to cherry-pick https://github.com/osbuild/osbuild-composer/pull/1725. Without it, the base tests on RHEL 9 fail.

**Review by thozza — APPROVED** [review]

**Pull request merged**